disperser/dataapi/ejector.go

```diff
@@ -86,11 +106,73 @@ func (e *ejector) eject(ctx context.Context, nonsigningRate *OperatorsNonsigning
 		return err
 	}
 
-	receipt, err := e.transactor.EjectOperators(ctx, operatorsByQuorum)
+	txn, err := e.transactor.BuildEjectOperatorsTxn(ctx, operatorsByQuorum)
 	if err != nil {
-		e.logger.Error("Ejection transaction failed", "err", err)
+		e.logger.Error("Failed to build ejection transaction", "err", err)
 		return err
 	}
+
+	var txID walletsdk.TxID
+	retryFromFailure := 0
+	for retryFromFailure < maxSendTransactionRetry {
+		gasTipCap, gasFeeCap, err := e.ethClient.GetLatestGasCaps(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get latest gas caps: %w", err)
+		}
+
+		txn, err = e.ethClient.UpdateGas(ctx, txn, big.NewInt(0), gasTipCap, gasFeeCap)
+		if err != nil {
+			return fmt.Errorf("failed to update gas price: %w", err)
+		}
+		txID, err = e.wallet.SendTransaction(ctx, txn)
+		var urlErr *url.Error
+		didTimeout := false
+		if errors.As(err, &urlErr) {
+			didTimeout = urlErr.Timeout()
+		}
+		if didTimeout || errors.Is(err, context.DeadlineExceeded) {
```

Comment:
So one timeout is from connecting to fireblocks endpoint, the other the fireblocks itself couldn't send txn internally?

Reply:
`didTimeout` detects whether it timed out from the http client, which can happen with fireblocks API call. `context.DeadlineExceeded` detects whether it timed out from from rpc calls, which can happen if wallet is using ethClient to send transactions